src/AppRoutes.js

```diff
@@ -77,9 +73,11 @@ const AppRoutes = (props) => {
       </Route>
       <Route
         path="/sign-in"
-        element={!authenticated ? <SignInView /> : <Navigate to={history?.state?.idx === 0 ? '/workspaces' : -1} />}
+        element={
+          !authenticated ? <SignInView /> : <Navigate to={providedUrlBeforeSignIn || providedUrl || '/workspaces'} />
+        }
       />
-      <Route path="/accessDenied" element={<AccessDeniedView />} />
+      <Route path="/accessDenied" element={!authenticated ? <SignInView /> : <AccessDeniedView />} />
```

Comment:
Could we add a fallback redirection if the user is authenticated AND authorized but they have targeted the AccessDenied URL ?